### ISSUE TITLE
Single GPU error

### ISSUE THREAD

**Issue by Virgilzz** [raoyongming/DenseCLIP#24]
Hi，I've modified the settings for single or multiple GPUs

norm_cfg = dict(type='BN', requires_grad=True)

But there were still such errors.
![Uploading image.png…]()
Does that mean I need to modify the training section in the mmseg source?

**Comment by Virgilzz** [comment]
[
![gitgubbb](https://user-images.githubusercontent.com/101557492/186604772-18d9f03e-1d70-4918-b7d5-2f9531f28b1b.png)
](url)

**Comment by raoyongming** [comment]
Hi @Virgilzz, thanks for your interest in our work.

It seems the error comes from the SyncBN layer. We convert all batch norm layers to SyncBN [here](https://github.com/raoyongming/DenseCLIP/blob/e1da471365402958b32c11b71b3aa4317de5af79/segmentation/train_denseclip.py#L149), which only supports DistributedDataParallel (DDP) environment. You may need to comment out this line if you want to train the model on a single GPU or without DDP.

**Comment by Virgilzz** [comment]
It has been solved. Thank you!
